src/prototype_room_basebuilder.js

```diff
@@ -67,7 +67,7 @@ Room.prototype.destroyStructure = function(structure) {
     structuresMin = 1;
   }
 
-  if (structures.length > structuresMin) {
+  if (structures.length > structuresMin && Room.structureIsEmpty(structure)) {
```

Comment:
Yeah, that's why I was asking about should I make this PR at all. This changes are will be very rare-used, may break something else, and any way, it's better if such cases would be monitored manualy) So I'm not insist this to be merged